### DISCUSSION TITLE
Tensor variables not read correctly in paraview?

### DISCUSSION THREAD

**Discussion by rebpoli** [libMesh/libmesh#4016] [Q&A] [answered]
Hi ,

I am adding the tensor components as variables, but paraview is not able to read them as a tensor.
I am exporting using ExodusII.
From what I recall, this used to work...

Any idea of what might be wrong?

![{A92729F0-517C-41AF-A11D-B6DEDC8AFA5A}](https://github.com/user-attachments/assets/ab1e0019-a47a-4fdc-882c-c7a78455892e)

**Comment by roystgnr** [comment]
Do they have to be added in a particular order?  https://discourse.paraview.org/t/automatic-naming-of-tensor-components/8762 suggests that you want to have YZ before XZ.

Otherwise I have no idea, but if you do hit upon something that works please let us know here.  I'd have imagined that I'd find an example of our solid mechanics module using that feature, but I couldn't...

**Reply by rebpoli** [reply]
Well .. the info is conflicting.

Other posts point to 
https://gitlab.kitware.com/vtk/vtk/blob/master/IO/Exodus/vtkExodusIIReader.cxx
(line 363)
`  const char endST23[] = "XXYYZZXYXZYZ";`

Anyway, I tried and it did not work either.

Do you know if the variables might be reordered in the export or libmesh flow?
How can I check that the ordering in the exodus file?

**Reply by roystgnr** [reply]
libMesh should be exporting variables in the same order as the `add_variable` calls that were made to the system (the first `add_variable` calls, in special cases - if you read your system from file then the file determines the ordering, and if you have redundant `add_variable` calls then then the first determines the ordering).

You can look at ExodusII file details by converting it from its underlying NetCDF3 or HDF5 format to ASCII and reading the output - there are utilities `ncdump` (available with NetCDF; with libMesh contrib/netcdf we install it for you alongside our own install) and `h5dump` (available with HDF5) to do so.

**Comment by rebpoli** [comment]
Thank you.

As a matter of fact, I tried it all just a few minutes ago.
The ordering is correct, as you mentioned.
Paraview is not able to collapse those vars into a tensor in any situation.
I asked the paraview community, let's see what they say.

Renato

**Comment by rebpoli** [comment, marked as answer]
Feedback from Paraview community.


*From:  Mathieu Westphal (Kitware) mwestphalDecember 3*

*Indeed, I confirm that the new IOSS exodus reader does not support tensors
yet.*

*You can use the legacy reader by loading the plugin:*

   - *Tools → ManagePlugins*
   - *Load “LegacyExodusReader”*
   - *File → Open → your file*
   - *Select “LegacyExodusReader”*

**Reply by roystgnr** [reply]
Thanks for inquiring into this!
